### DISCUSSION TITLE
Information on instance sleeping?

### DISCUSSION THREAD

**Discussion by everyplace** [community/community#4230] [Codespaces] [answered]
I haven’t been able to find this in the docs, but does anyone have any information on the current sleep timer of the codespace instances we create? 

For example, if I’m running an codespace that exposes a port, and then I want to run a separate app to run tests against the exposed port: 
- How long will the first codespace run? Is it for as long as I have the space actively open in my browser? 
- Will it auto-sleep and reboot like how Glitch handles unboosted instances? 
- What if I spin up a codespace, start running tests against it (again, externally) then shut the original codespace window?

 I know these are all strange uses, but it has to be documented *somewhere*, right?

**Comment by tanmayeekamath** [comment, marked as answer]
Hey @everyplace, the current default suspend timeout is 30 mins. As long as you are interacting with the Codespace (or the port where your app is running) it would not count as idle. If you navigate away from the browser tab/running app; then that will count towards idle activity and after 30 mins of idle activity we will suspend your Codespace. 

> How long will the first codespace run? Is it for as long as I have the space actively open in my browser?

That is correct. 

> Will it auto-sleep and reboot like how Glitch handles unboosted instances?

I'm not familiar with Glitch, for Codespaces the reconnect will drop you into the same state that you left it in when it was suspended. 

> What if I spin up a codespace, start running tests against it (again, externally) then shut the original codespace window? 

If you are accessing a forwarded port from within the Codespace, that should still count as activity. So we would not expect your Codespace to suspend. 


Thanks for your feedback. we have plans to soon add this to our docs as well. :) 

Hope this helps!

**Reply by everyplace** [reply]
This is a great response, thank you so much for the detailed answers.
